src/core_plugins/kibana/public/discover/embeddable/search_embeddable_factory.js

```diff
@@ -74,9 +74,6 @@ export class SearchEmbeddableFactory extends EmbeddableFactory {
         const rootNode = angular.element(domNode);
         rootNode.append(searchInstance);
 
-        // TODO: There is probably a better way to do this.
-        searchScope.$on('ready:vis', () => this.courier.fetch());
```

Comment:
If there was just searches on a dashboard, that was required to queue a `courier.fetch()` otherwise the dashboard would never trigger a courier run.

This was replaced by a `courier.fetch()` in the place the doc_table did emit that event earlier. This is not an ideal solution, since the doc_table currently has two modes: either passing in a `searchSource` that it uses to execute the search (only happens when in dashboard) or passing in `hits` because the using place takes care of loading that data (as Discover does).

In the long run, the doc_table should just be that simple showing `hits` solution, so it currently requesting it's own fetch is just a workaround that replaces this call to `courier.fetch`.